Mismatch type for percent when compiling for `thumbv7em-none-eabihf`

When compiling a particular slint code on `thumbv7em-none-eabihf` and including it with the `include_modules!()` macros, compilation fails.

cargo.toml dependency

```toml
slint = { version = "0.3.1", default-features = false, features = ["compat-0-3-0", "unsafe-single-threaded", "libm"] }

```


Minimum slint code to reproduce.
```slint
Inner := Rectangle { 
    property <percent> val;
    background: val < 50% ? red : blue;
}

// It seems the indirection is required
Outer := VerticalLayout {
    property <percent> val;
    Inner { val: parent.val;    } // Commenting this line does not produce a compiler error

    Rectangle { background: val < 50% ? green : yellow; }

} 

export AppWindow := Window {
    width: 240px;
    height: 240px;
    Outer { val: 50%; }
}
```

Rust error output
```console
error[E0308]: mismatched types
  --> /home/xgroleau/Documents/wearable/target/thumbv7em-none-eabihf/debug/build/firmware-app-f43eb8afe1782cb3/out/main.rs:61:55
   |
61 |                     ) . apply_pin (_self) . get ()) < (50f64)) {
   |                                                       ^^^^^^^ expected `f32`, found `f64`

error[E0277]: can't compare `f32` with `f64`
  --> /home/xgroleau/Documents/wearable/target/thumbv7em-none-eabihf/debug/build/firmware-app-f43eb8afe1782cb3/out/main.rs:61:53
   |
61 |                     ) . apply_pin (_self) . get ()) < (50f64)) {
   |                                                     ^ no implementation for `f32 < f64` and `f32 > f64`
   |
   = help: the trait `PartialOrd<f64>` is not implemented for `f32`
   = help: the following other types implement trait `PartialOrd<Rhs>`:
             <f32 as PartialOrd<fixed::FixedI128<Frac>>>
             <f32 as PartialOrd<fixed::FixedI16<Frac>>>
             <f32 as PartialOrd<fixed::FixedI32<Frac>>>
             <f32 as PartialOrd<fixed::FixedI64<Frac>>>
             <f32 as PartialOrd<fixed::FixedI8<Frac>>>
             <f32 as PartialOrd<fixed::FixedU128<Frac>>>
             <f32 as PartialOrd<fixed::FixedU16<Frac>>>
             <f32 as PartialOrd<fixed::FixedU32<Frac>>>
           and 146 others

Some errors have detailed explanations: E0277, E0308.
For more information about an error, try `rustc --explain E0277`.
```
```sh
❯ cargo --version
cargo 1.66.0-nightly (071eeaf21 2022-10-22)
```

--- a/internal/compiler/generator/rust.rs
+++ b/internal/compiler/generator/rust.rs
@@ -1829,6 +1829,7 @@ fn compile_expression(expr: &Expression, ctx: &EvaluationContext) -> TokenStream
                             | Type::PhysicalLength
                             | Type::LogicalLength
                             | Type::Angle
+                            | Type::Percent
                     ) =>
                 {
                     (Some(quote!(as f64)), Some(quote!(as f64)))

--- a/tests/cases/issues/issue_1837_percent_comparison.slint
+++ b/tests/cases/issues/issue_1837_percent_comparison.slint
@@ -1,0 +1,22 @@
+// Copyright © SixtyFPS GmbH <info@slint-ui.com>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-commercial
+
+Inner := Rectangle {
+    property <percent> val;
+    background: val < 50% ? red : blue;
+}
+
+// It seems the indirection is required
+Outer := VerticalLayout {
+    property <percent> val;
+    Inner { val: parent.val;    } // Commenting this line does not produce a compiler error
+
+    Rectangle { background: val < 50% ? green : yellow; }
+
+}
+
+export AppWindow := Window {
+    width: 240px;
+    height: 240px;
+    Outer { val: 50%; }
+}

